orchestrator/services/processes.py

```diff
@@ -183,8 +191,15 @@ def _db_log_step(
         executed_at.append(str(current_step.executed_at))
 
         # write new state info and execution date
-        current_step.state = {**step_state, "retries": retries, "executed_at": executed_at}
-        logger.info("Updating existing process step with state info about the error", retries=retries)
+        current_step.state = {
+            **step_state,
```

Comment:
Since Python 3.9 there is a cleaner way to merge dictionaries with the | operator